go/get_all_records_test.go

```diff
@@ -0,0 +1,374 @@
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Network.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PIFMetrics.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	if session.APIVersion >= xenapi.APIVersion2_21 {
+		_, err = xenapi.Observer.GetAllRecords(session)
+		if err != nil {
+			t.Log(err)
+			t.Fail()
+			return
+		}
+	}
+	_, err = xenapi.PBD.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PCI.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PGPU.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PIFMetrics.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PIF.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PoolPatch.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PoolUpdate.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Pool.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PUSB.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PVSCacheStorage.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PVSProxy.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PVSServer.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.PVSSite.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	if session.APIVersion >= xenapi.APIVersion2_21 {
+		_, err = xenapi.Repository.GetAllRecords(session)
+		if err != nil {
+			t.Log(err)
+			t.Fail()
+			return
+		}
+	}
+	_, err = xenapi.Role.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.SDNController.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Secret.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.SM.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.SR.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Subject.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Task.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.Tunnel.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	_, err = xenapi.USBGroup.GetAllRecords(session)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	// XAPI Message Removed
```

Comment:
I would recommend removing rather that commenting code out to keep the files clean.

(Out of curiosity, what was wrong with it?)

Reply:
I'd like to test all "get_all_records" method for each XAPI data model class. But for some XAPI class, like VMPP, it returns XAPI error "XAPI Message Removed". In page https://xapi-project.github.io/xen-api/classes/vmpp.html, you can find this method for VMPP is removed from XS6.2. So, I just comment out this call. But year, I can clean those removed call in this file.

Reply:
Aah yes, I forgot, VMPP was deprecated in favour of VMSS, there are a few more deprecated classes like Pool_patch etc. We don't need to test those (they don't have valid vlaues in newer servers anyway).